### PR TITLE
Calc: fix formulabar expander being cropped on mobile

### PR DESCRIPTION
**Pull request by pedropintosilva**
commit db6581582a46fa0ffb1f671babbbec41d7c62e04 (HEAD -> private/pedro/fix-mobile-calc-expander-hide-startformula, origin/private/pedro/fix-mobile-calc-expander-hide-startformula)
Author: Pedro Pinto Silva <pedro.silva@collabora.com>
Date:   Tue Mar 7 09:46:42 2023 +0100

    Mobile: calc: formulabar element: align to top on expanded
    
    Signed-off-by: Pedro Pinto Silva <pedro.silva@collabora.com>
    Change-Id: I1579f526daa18c45af0e7106eff43d5491aaf89b

commit 164772259e1800afde50d26fbf5f1817b7433e3c
Author: Pedro Pinto Silva <pedro.silva@collabora.com>
Date:   Mon Mar 6 12:26:37 2023 +0100

    Mobile: calc: Fix addressinput  misalignment
    
    In some browsers addressinput was misaligned (no initial margin)
      - Remove FF specific margin-left
      - Add non LTR specific margin to every browser
      - Use the same dimension (8px) as used in the .formulabar.toolbox's
      grid-column-gap
    
    Signed-off-by: Pedro Pinto Silva <pedro.silva@collabora.com>
    Change-Id: I6a2e86a9a56f05bbf514d12a080fbcb2408cb816

commit 8e0b45fc36cad59e528b20ec4277c806fb175955
Author: Pedro Pinto Silva <pedro.silva@collabora.com>
Date:   Fri Mar 3 18:34:19 2023 +0100

    Calc: fix formulabar expander being cropped on mobile
    
    - Make sure all elements within formulabar toolbox fit even
    on smaller devices
        - Do not user margins to position, use instead grid
    - Remove Equal sign (startformula). Before this commit the
    startformula icon was visible and then invisible once the user
    enter the cell to type -> it seems this was initially intended to save
    space but the truth of the matter is that it causes a flicker and
    where the position of the elements shift making even harder to write
    on mobile.
        - Solution: do not create that element in the first place on
    mobile. Plus when the user enter the cell can always press the "="
    right via the native keyboard anyhow.
    
    note: avoid multiple instances of the same condition in FormulaBarJSDialog
    
    Signed-off-by: Pedro Pinto Silva <pedro.silva@collabora.com>
    Change-Id: Ic4f3b7fddb9d862170a0e95243f4b19a89340575

### CODE REVIEW

File: browser/src/control/Control.FormulaBar.js [hunk 1]
```diff
@@ -141,16 +141,19 @@ L.Map.include({
 	onFormulaBarFocus: function() {
 		var mobileTopBar = w2ui['actionbar'];
 		var jsdialogFormulabar = this.formulabar;
+		var target = jsdialogFormulabar;
 
-		var target = window.mode.isMobile() ? mobileTopBar : jsdialogFormulabar;
+		if (window.mode.isMobile() === true) {
+			mobileTopBar.hide('undo');
+			mobileTopBar.hide('redo');
+			$('#formulabar').addClass('formulaBarFocus');
```
Comment (eszkadev) [inline browser/src/control/Control.FormulaBar.js:149]: here class is set but never removed

Comment (eszkadev) [reply browser/src/control/Control.FormulaBar.js:149]: also not used anywhere

Comment (pedropintosilva) [reply browser/src/control/Control.FormulaBar.js:149]: ah forgot to remove that

File: browser/src/control/Control.FormulaBarJSDialog.js [hunk 2]
```diff
@@ -47,23 +47,19 @@ L.Control.FormulaBarJSDialog = L.Control.extend({
...
-					{
-						id: 'startformula',
-						type: 'customtoolitem',
-						text: _('Formula'),
-					},
-					// on mobile we show other buttons on the top bar
-					(!window.mode.isMobile()) ? (
+						},
 						{
+							id: 'startformula',
```
Comment (eszkadev) [inline browser/src/control/Control.FormulaBarJSDialog.js:52]: now it doesn't work on desktop (all icons are always hidden)

File: browser/src/control/Control.FormulaBarJSDialog.js [hunk 3]
```diff
@@ -32,61 +32,79 @@ L.Control.FormulaBarJSDialog = L.Control.extend({
...
+						{
+							id: 'functiondialog',
+							type: 'toolitem',
+							text: _('Function Wizard'),
+							command: '.uno:FunctionDialog'
+						}, {
+							id: 'sc_input_window',
+							type: 'multilineedit',
+							text: text ? text : '',
+							rawKeyEvents: window.mode.isDesktop() ? true : undefined,
```
Comment (eszkadev) [inline browser/src/control/Control.FormulaBarJSDialog.js:97]: you can remove these ifs now, here it is alwasy mobile

Comment (pedropintosilva) [reply browser/src/control/Control.FormulaBarJSDialog.js:97]: ah good catch!

File: browser/src/control/Control.FormulaBarJSDialog.js [hunk 4]
```diff
@@ -32,61 +32,79 @@ L.Control.FormulaBarJSDialog = L.Control.extend({
...
-					}
-				]
-			}
-		];
+						},
+						{
+							id: 'sc_input_window',
+							type: 'multilineedit',
+							text: text ? text : '',
+							rawKeyEvents: window.mode.isDesktop() ? true : undefined,
```
Comment (eszkadev) [inline browser/src/control/Control.FormulaBarJSDialog.js:73]: remove condition here it is always non mobile

Comment (pedropintosilva) [reply browser/src/control/Control.FormulaBarJSDialog.js:73]: what about tablet? shouldn't we leave those conditions for seTextInput and rawKeyEvents in this case?

File: browser/src/control/Control.FormulaBarJSDialog.js (hunk 3 above)
Comment (pedropintosilva) [reply browser/src/control/Control.FormulaBarJSDialog.js]: removed

File: browser/src/control/Control.FormulaBarJSDialog.js (hunk 4 above)
Comment (eszkadev) [reply browser/src/control/Control.FormulaBarJSDialog.js:73]: ah, yes you are right. it should be as is here

**Review by eszkadev — APPROVED** [review]

**Comment by pedropintosilva** [comment]
cypress CI seems passing but then it goes :zombie: on me... so I don;'t know. Rebasing with master and force pushing ..maybe it helps

**Comment by pedropintosilva** [comment]
```
clicat: write error: Resource temporarily unavailable
Makefile:683: recipe for target '/home/collabora/jenkins/workspace/github_online_master_debug_vs_co-22.05/cypress_test/workdir/track/mobile/calc/hamburger_menu_spec.js.done' failed
make[2]: *** [/home/collabora/jenkins/workspace/github_online_master_debug_vs_co-22.05/cypress_test/workdir/track/mobile/calc/hamburger_menu_spec.js.done] Error 1
make[2]: *** Waiting for unfinished jobs....
Build timed out (after 66 minutes). Marking the build as aborted.
Build was aborted
[PostBuildScript] - [INFO] Executing post build scripts.
[github_online_master_debug_vs_co-22.05] $ /bin/bash /tmp/jenkins13483699053413911926.sh
INFO : AFTER ABORT or FAILURE CLEAN UP STARTED
INFO : Running `git clean -ffd` ...
INFO : Running `make distclean` ...
INFO : Cleaning stuck jails on the slave if there is any
INFO : Cleaning hanged build processes on the slave if there is any
Makefile:5629: recipe for target 'check-recursive' failed
make: *** [check-recursive] Killed
INFO : Cleaning zombie cypress and chromium headless processes if there is any
Makefile:683: recipe for target '/home/collabora/jenkins/workspace/github_online_master_debug_vs_co-22.05/cypress_test/workdir/track/mobile/calc/sheet_operation_spec.js.done' failed
make[2]: *** [/home/collabora/jenkins/workspace/github_online_master_debug_vs_co-22.05/cypress_test/workdir/track/mobile/calc/sheet_operation_spec.js.done] Killed
make[2]: Leaving directory '/home/collabora/jenkins/workspace/github_online_master_debug_vs_co-22.05/cypress_test'
Makefile:538: recipe for target 'check-am' failed
make[1]: *** [check-am] Error 2
make[1]: Leaving directory '/home/collabora/jenkins/workspace/github_online_master_debug_vs_co-22.05/cypress_test'
INFO : CLEANING COMPLETE
[PostBuildScript] - [INFO] Build does not have any of the results [SUCCESS]. Did not execute build step #1.
Setting status of 371bd24a8928e47ab7709d63e39595f836af09a0 to FAILURE with url https://cpci.cbg.collabora.co.uk:8080/job/github_online_master_debug_vs_co-22.05/3832/ and message: 'FAILURE
 '
Using context: Unit & Cypress & Notebookbar Tests against Core co-22.05
Finished: ABORTED

```

**Pull request merged**